gql/dsl.py

```diff
@@ -117,29 +102,32 @@ def __str__(self):
         return print_ast(self.ast_field)
 
 
-def field(field, **args):
+def selection_field(field, **args):
     if isinstance(field, GraphQLField):
+        # TODO: Pass name argument
```

Comment:
Can you add a test that passes a GraphQLField to select(), to get that case covered?

Reply:
No problem!

Reply:
The init function at DSLField requires it:
```python
    def __init__(self, name, field):
        self.field = field
        self.ast_field = ast.Field(name=ast.Name(value=name), arguments=[])
        self.selection_set = None
```
I tried this query:
```
def test_pass_grapqhl_field_on_query(ds):
    result = ds.query(
        ds.Query.hero.select(
            queryType.fields.get('hero')
        )
    )
    expected = {
        'hero': {
            'name': 'R2-D2'
        }
    }
    assert result == expected
```

But no success :/ @Cito

Reply:
Right, a bare GraphQLField has no name. Not sure how this was intended to be used. Maybe we should simply remove that whole if branch and the `**args`.

Regarding the missing coverage in requests. The "raise Value" happens e.g. when the server returned JSON that is not an object, but an array. The "requests.HTTPError" happens when the server does not report an valid GraphQL result, but with a status of OK.

Reply:
Sounds good! I will provide last two missing test on another PR.

Reply:
Ok. Is this PR ready to be merged then from your side?

Reply:
Yeah! Ready for merge!